index.bs

```diff
@@ -643,6 +643,21 @@ partial interface Gamepad {
 <dfn attribute for="Gamepad" id="gamepad-getvrdisplays-attribute">displayId</dfn>
 Return the {{VRDisplay/displayId}} for the associated {{VRDisplay}}.
 
+# Processing Model # {#proc}
+
+## submitFrame ## {#submitframe-proc}
+
+The are are multiple situations where applications won't invoke submitFrame at the expected rate (the VRDisplay native framerate). The User Agent must intervene to protect the user from motion sickness or to consistently inform her about different situations like errors or navigation between pages. This is the breakdown of the different scenarios:
+
+* The application calls submitFrame at the expected rate but a few frames are dropped. The User Agent will re-project the last submitted frame.
+
+* The application calls submitFrame consistently below the expected framerate to a degree that could make the user feel umconfortable. The User agent will show a semi transparent overlay on top of the application with a notification icon (an hourglass for instance) until the application improves performance. If the low framerate persists for a long time the User Agent will show an opaque overlay indicating that the application is too slow to be presented.
+
+* An application can stop calling submitFrame all toghether under several circumstances:
+    * A content exception or error has ocurred. In this case the User Agent will notify the user with an opaque overlay.
+    * The User Agent transitions to a different site that will eventually call submitFrame (the User Agent will statically determine if a site will call submitFrame by the presence of meta name="vr"). The User Agent will render an opaque overlay with a loading spinner to inform the user of the situation. Once the new site starts calling submitFrame a semi transparent overlay with the new URL will be shown to notify the user of the site transition.
```

Comment:
The meta tag sounds good in practice, but fails on modern content. The meta tag MAY be in an iframe, not in the top level page. We still need the page to attach to onvrdisplayactivate since they can't otherwise call requestPresent or submitFrame. 

To me the largest bother in the current text is that it implies a page can call submitFrame. Which it can't. During a navigation, the page MUST hook onvrdisplayactivate, then call getVRDevices, then call requestPresent IN THE CONTEXT of the onvrdisplayactivate (user intent, provided by the browser because it can detect the user navigated while in VR) and finally, after it does a bunch of other stuff, call submitFrame. We should reword the section to clarify what the target of a VR navigation would actually do and not imply that they can call submitFrame directly.

And a browser would still need some timeout, interstitial and user controls. (a combination of them all maybe) since it is always possible for the content not to achieve frame submission for any number of reasons. The annotations of meta tags can simply increase thee confidence that the browser should wait longer before recommending an alternative to a user. Today we say, "Hey this web page isn't responding", and thats annoying on some pages that are just taking a long time, but handy in others. Its a heuristic. So we should provide some minimum set of guarantees on time to wait and then allow browsers to make those longer via other means if they choose (such as a history entry for the page saying that it did indeed make it into VR on a previous navigation).